backend-python/src/telestion/backend/config.py

```diff
@@ -0,0 +1,110 @@
+import argparse
+import json
+import os
+from pathlib import Path
+from typing import Any, TypeVar
+
+from pydantic import BaseModel, Field, ConfigDict
+
+
+class TelestionConfig(BaseModel):
+    dev: bool = False
+    nats_url: str = Field(alias="NATS_URL")
+    nats_user: str | None = Field(alias="NATS_USER", default=None)
+    nats_password: str | None = Field(alias="NATS_PASSWORD", default=None)
+    config_file: Path | None = Field(alias="CONFIG_FILE", default=None)
+    config_key: str | None = Field(alias="CONFIG_KEY", default=None)
+    service_name: str = Field(alias="SERVICE_NAME")
+    data_dir: Path = Field(alias="DATA_DIR")
+
+    unparsed_cli: list[str] = Field(alias="_telestion_validator_unparsed_cli")
+
+    # To include all envs and config parts -> it is recommended to add a custom subtype
+    model_config = ConfigDict(
+        extra='allow'
+    )
+
+
+# With this we allow users to extend TelestionConfig for finer control over custom config fields
+_TelestionConfigT = TypeVar("_TelestionConfigT", bound=TelestionConfig)
+
+
+def build_config(clazz: type[_TelestionConfigT] = None) -> _TelestionConfigT:
+    if clazz is None:
+        clazz = TelestionConfig
```

Comment:
Because in Python function default arguments are retained between executions. Yes, that's a feature, not a bug.

https://docs.python.org/3/faq/programming.html#why-are-default-values-shared-between-objects